src/Polly/Caching/AsyncCacheEngine.cs

```diff
@@ -43,7 +45,7 @@ internal static class AsyncCacheEngine
         if (cacheHit)
```

Comment:
Is there a way to annotate it so the compiler knows that if `cacheHit` is `true` then `valueFromCache` is non-null?

Reply:
Or, are there use cases where `null` is a valid value in a cache (e.g. _"I've looked this up and it doesn't exist, so cache that fact"_), and we should therefore allow null?